### DISCUSSION TITLE
Confirm version for production branch / how to update beyond 1.17?

### DISCUSSION THREAD

**Discussion by sirkelsalot** [immich-app/immich#487] [Q&A] [answered]
I'm running in docker-compose on Unraid after pulling down the compose and env files from the main page. I've merge those into my regular compose file and things are generally ok.

My issue is trying to understand how to get a the more recent releases? It appears this approach is still on server version 1.17 so I am receiving the warning on both server and mobile that this is an outdated instance. Mobile is version 1.22 build 32.

Aside from the warnings on both, now on mobile i can no longer able to see any photos, but they appear fine in the web version. I assume this is caused by the version gap from mobile/server.

So my question is, is it expected to be stuck on 1.17 with the production branch? Is there a different recommended deployment config to stay more current on the server?

**Comment by alextran1502** [comment, marked as answer]
Can you make sure to use the latest `docker-compose` file with the `release` tag for all Immich's images?

Then you will need to perform `docker-compose pull` in order to get the latest update.

**Reply by sirkelsalot** [reply]
Correct. Using the 'release' tag for all containers. 

I followed the instructions as shown on the main page/readme (below). And it still shows server version 1.17 after a fresh install. Full compose listed at the bottom, for reference. I've tried the steps a few times with new directories, same results.

One thing I noticed when I followed the instructions below, I received an error saying VITE_SERVER_ENDPOINT is required. I populated that in the .env to move forward. But I did see a post that that should not longer be required (and presumably the reason it is not longer in the .env example), but based on the version I'm pulling down it is still required.

![image](https://user-images.githubusercontent.com/2467374/185445390-6b297b4a-0594-4985-ab26-f92ac703aab3.png)


> Step 1 - Download necessary files
> 
>     Create a directory called immich-app and cd into it.
> 
>     Get docker-compose.yml
> 
> wget https://raw.githubusercontent.com/immich-app/immich/main/docker/docker-compose.yml
> 
>     Get .env
> 
> wget -O .env https://raw.githubusercontent.com/immich-app/immich/main/docker/.env.example
> 
> Step 2 - Populate .env file with custom information
> 
> [See the example .env file](https://github.com/immich-app/immich/blob/main/docker/.env.example)
> 
>     Populate custom database information if necessary.
>     Populate UPLOAD_LOCATION as prefered location for storing backup assets.
>     Populate a secret value for JWT_SECRET, you can use this command: openssl rand -base64 128
>     [Optional] Populate Mapbox value to use reverse geocoding.
> 
> Step 3 - Start the containers
> 
>     Run docker-compose up or docker compose up (based on your docker's version)
> 
> Step 4 - Register admin user
> 
>     Navigate to the web at http://<machine-ip-address>:2283 and follow the prompts to register admin user.

```
version: "3.8"

services:
  immich-server:
    image: altran1502/immich-server:release
    entrypoint: ["/bin/sh", "./start-server.sh"]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    environment:
      - NODE_ENV=production
    depends_on:
      - redis
      - database
    restart: always

  immich-microservices:
    image: altran1502/immich-server:release
    entrypoint: ["/bin/sh", "./start-microservices.sh"]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    environment:
      - NODE_ENV=production
    depends_on:
      - redis
      - database
    restart: always

  immich-machine-learning:
    image: altran1502/immich-machine-learning:release
    entrypoint: ["/bin/sh", "./entrypoint.sh"]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    environment:
      - NODE_ENV=production
    depends_on:
      - database
    restart: always

  immich-web:
    image: altran1502/immich-web:release
    entrypoint: ["/bin/sh", "./entrypoint.sh"]
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich_redis
    image: redis:6.2
    restart: always

  database:
    container_name: immich_postgres
    image: postgres:14
    env_file:
      - .env
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
      PG_DATA: /var/lib/postgresql/data
    volumes:
      - pgdata:/var/lib/postgresql/data
    restart: always

  immich-proxy:
    container_name: immich_proxy
    image: altran1502/immich-proxy:release
    ports:
      - 2283:80
    logging:
      driver: none
    depends_on:
      - immich-server
    restart: always

volumes:
  pgdata:

**Reply by alextran1502** [reply]
The instruction will bring up the container.

To update the already running instance, please run

`docker-compose down && docker-compose pull`

Then start then up again

`docker-compose up -d`

Those commands will

1. Bring down the running containers
2. Pull the latest update
3. Start the containers with latest update

**Reply by sirkelsalot** [reply]
Ok. I think that did the trick. Here's where I may have gone wrong.

- Ignorance assuming that starting from scratch would always pull the absolute latest. I thought I correctly provisioned several brand new deployments following the readme instructions, but always got v1.17. 
- I had in old notes to use one of the following docker-compose commands for updates, which did not pull down the latest
```
docker-compose up --detach --build --remove-orphans
or
docker-compose up --force-recreate --build -d --remove-orphans
```

- Your suggested command below worked to get me on the latest
`docker-compose down && docker-compose pull`
